### PR TITLE
Kiota based Python sdk

### PR DESCRIPTION
**Pull request by andreaTP**
This is a WIP Kiota generation of the Python SDK for Registry.

### CODE REVIEW

File: python-sdk/tests/basic_test.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            "title": "My API",
+            "version": "1.0.0"
+        },
+        "paths": {}
+    }"""
+    create_artifact = await client.groups_by_id("default").artifacts.post(payload)
+    assert create_artifact.id is not None
+    
+    return_artifact = await client.groups_by_id("default").artifacts_by_id(create_artifact.id).get()
+    assert json.loads(str(return_artifact, "utf-8").replace("'", '"')) == json.loads(payload.content)
```
Comment (andreaTP) [inline python-sdk/tests/basic_test.py:70]: the `replace` here should not happen, I'll check tomorrow where the bug is happening.

**Comment by paoloantinori** [comment]
❤️

**Review by EricWittmann — APPROVED** [review]
It's been awhile since I've done any python programming, but overall this seems reasonable to me.  I suppose we don't actually need to use python to build the python SDK, since it's just making an os system call to the kiota CLI.  Can the kiota maven plugin be used to generate non-java SDKs?  If yes, it might be easier to ensure this works in our downstream builds when that becomes a requirement.

**Comment by andreaTP** [comment]
Thanks a lot for the comment @EricWittmann ! That's an interesting take that I haven't thought about in first place, though I still think this PR makes sense in it's current incarnation, let me explain the reasons.

We can use the Maven plugin to generate the Python code, but the plugin will soon fall short at the next steps: testing, linting, packaging and publishing will require extra research and development to use non-standard tooling for the language.

I also decided to go "full-in" into a "python way of doing it" as an attempt to reduce the friction for possible contribution from python developers.

I think those reasons are compelling enough to justify the current approach but, please, let me know if I'm missing anything!

**Comment by EricWittmann** [comment]
I think those are good arguments.  The consequence is that we'll need to figure out how to productize this when the time comes.  Since this will almost certainly be used by RHOAD, they will (presumably) need a product version of this.  So we'll need to explore how to make that happen (when we need it).

For now, I think this makes sense.

Thanks!

**Review by EricWittmann — APPROVED** [review]

**Comment by EricWittmann** [comment]
@andreaTP please feel free to merge at your convenience. :)

**Pull request merged**